topics/go/algorithms/data/heap/heap.go

```diff
@@ -0,0 +1,158 @@
+// Package heap implements a heap interface, and add the ability to store values
+// of type Data in the minimum Heap.
+package heap
+
+import (
+	"errors"
+)
+
+// Data represents what is being stored on the heap.
+type Data struct {
+	Value int
+	Index int
+}
+
+// Heap represents a list of data.
+type Heap struct {
+	capacity int
```

Comment:
SInce you are not changing this when heap memory is allocated, can't we just use the cap function?